python_modules/libraries/dagster-dbt/dagster_dbt/asset_utils.py

```diff
@@ -956,6 +972,18 @@ def get_asset_check_key_for_test(
     )
 
 
+def get_checks_on_sources_upstream_of_selected_assets(
+    assets_def: AssetsDefinition, selected_asset_keys: AbstractSet[AssetKey]
+) -> AbstractSet[AssetCheckKey]:
+    sources_upstream_of_selected_assets = {
+        dep.asset_key for key in selected_asset_keys for dep in assets_def.specs_by_key[key].deps
```

Comment:
my brain is too small for these nested for loops, but I trust this does the right thing. could we make this even simpler by just replacing this with `assets_def.keys_by_input_name.values()`? I think that implementation does the same general thing under the hood but better to centralize the logic

Reply:
I see your point but i think there's subtlety here do to the "global asset keys" vs "computation asset keys" dichotomy. keys_by_input_name relies on AssetsDefinition.keys which is resolved from the computation asset keys, which isn't what we want here i think